makeBuild.py

```diff
@@ -129,22 +123,24 @@ def build_book(book):
     print("---------------------------------------")
 
     for book in os.listdir(os.path.join("drupal-build", distro)):
-
-        #print(os.getcwd() + "\n")
         # skip any non-directory entries
-        if not os.path.isdir("drupal-build/" + distro + "/" + book):
+        if not os.path.isdir(os.path.join("drupal-build", distro, book)):
             continue
+
         # rest api book is a pain and doesn't convert well
         if book == "rest_api":
             continue
 
-        if os.path.exists(os.path.join("drupal-build", distro, book,"hugeBook.flag")):
-            for secondary_book in os.listdir(os.path.join("drupal-build", distro, book)):
-                if os.path.isdir("drupal-build/" + distro + "/" + book + "/" + secondary_book):
-                    all_validated = all_validated and build_book(book+"/"+secondary_book)
-        else:
-            all_validated = all_validated and build_book(book)
-if not all_validated:
-    sys.exit(-1)
-else:
-  print("All Successful")
+        try:
+            if os.path.exists(os.path.join("drupal-build", distro, book, "hugeBook.flag")):
+                for secondary_book in os.listdir(os.path.join("drupal-build", distro, book)):
+                    if os.path.isdir(os.path.join("drupal-build", distro, book, secondary_book)):
+                        build_book(book + "/" + secondary_book)
+            else:
+                build_book(book)
+
+        except Exception as e:
+            print(f"Error building {book}: {e}")
+
+        finally:
+            print("Build succeeded")
```

Comment:
As far as I could work out, your changes are intended to save time/resources in case a book fails the build, creating an immediate stop instead of continuing builds, is this correct?

If it is: I am not sure whether this is the best approach. If errors exist in several books, isn't it better for the writer to know about all of them? On the other hand, the CI will end faster if a failure does exist. I'm really unsure.

(Separately you introduce os.path.join() in several places - that is quite useful and I hav no questions about it, thanks!)

Reply:
Yeah not sure tbh. For me, I think failing fast and first is good. Will need to get other opinions.